lib/msf/core/auxiliary/masscan.rb

```diff
@@ -0,0 +1,277 @@
+# -*- coding: binary -*-
+require 'rex/parser/masscan_nokogiri'
+require 'rex/parser/masscan_xml'
+require 'open3'
+
+module Msf
+
+###
+#
+# This module provides methods for interacting with nmap.
+# Modules that include this should define their own nmap_build_args()
+# function, and usually should have some method for dealing with
+# the data yielded from nmap_hosts(). See auxiliary/scanner/oracle/oracle_login
+# for an example implementation.
+#
+###
+
+module Auxiliary::Masscan
+
+attr_accessor :massscan_args, :masscan_bin, :masscan_log
+attr_reader :masscan_pid, :masscan_ver
+
+def initialize(info = {})
+  super
+
+  register_options([
+    OptAddressRange.new('RHOSTS', [ true, "The target address range or CIDR identifier"]),
+    OptBool.new('NMAP_VERBOSE', [ false, 'Display nmap output', true]),
+    OptString.new('RPORTS', [ false, 'Ports to target']), # RPORT supersedes RPORTS
+  ], Auxiliary::Nmap)
```

Comment:
I don't believe this should be `Auxiliary::Nmap`.  You could simply remove this last optional argument or specify `Auxiliary::Masscan`